sympy/concrete/expr_with_limits.py

```diff
@@ -25,7 +26,18 @@ def _common_new(cls, function, *symbols, **assumptions):
     both ExprWithLimits and AddWithLimits."""
     function = sympify(function)
 
-    if hasattr(function, 'func') and isinstance(function, Equality):
+    if isinstance(function, Equality):
+        # This transforms e.g. Integral(Eq(x, y)) to Eq(Integral(x), Integral(y))
+        # but that is only valid for definite integrals.
+        limits, orientation = _process_limits(*symbols)
+        if not (limits and all(len(limit) == 3 for limit in limits)):
+            SymPyDeprecationWarning(
+                feature='Integral(Eq(x, y))',
```

Comment:
This deprecation isn't triggering for definite integrals. `Integral(Eq(x, y))` and `Integral(Eq(x, y), x)` raise, but `Integral(Eq(x, y), (x, 0, 1))` does not. Neither do `Sum` or `Product` since they are always definite.

I will fix this in https://github.com/sympy/sympy/pull/22900, but we may want to extend the deprecation period as a result.

Reply:
I guess this was intentional. I didn't notice the `if` conditions. IMO Eq shouldn't ever work (that should be something an Equation class does), but I guess since it currently works we can leave it.

Reply:
I don't remember why this`if` condition is there. There might not be a good reason.

Reply:
Actually definite integrals are at least well defined e.g. if `a=b` then equality still holds for a definite integral of both sides. The same is not true of indefinite integrals because they can differ by a constant. I guess that's where this came from.

Reply:
Yeah I thought this was about not letting Eq be an expression but it's only about ambiguity with indefinite integrals. I'll clarify this when I update the docs for this deprecation.